src/oneD/StFlow.cpp

```diff
@@ -1036,14 +1044,14 @@ void StFlow::evalContinuity(size_t j, double* x, double* rsd, int* diag, double
     //
     //    d(\rho u)/dz + 2\rho V = 0
     //----------------------------------------------
-    if (!m_isFree) {
+    if (m_usesLambda) {
         // Note that this propagates the mass flow rate information to the left
         // (j+1 -> j) from the value specified at the right boundary. The
         // lambda information propagates in the opposite direction.
         rsd[index(c_offset_U,j)] =
             -(rho_u(x,j+1) - rho_u(x,j))/m_dz[j]
             -(density(j+1)*V(x,j+1) + density(j)*V(x,j));
-    } else {
+    } else if (m_isFree) {
```

Comment:
I think there is some further simplification possible here, given that in the `m_isFree` case $V=0$.